apiserver/facades/client/sshclient/shim.go

```diff
@@ -148,26 +141,61 @@ func (b *backend) GetMachineForEntity(tagString string) (SSHMachine, error) {
 	}
 }
 
-// UnitVirtualPublicHostKey calls the underlying UnitVirtualHostKey state method
-// and encodes the result into a PEM string.
-func (b *backend) UnitVirtualPublicHostKeyPEM(unitID string) (string, error) {
-	// The keys are persisted PEM encoded.
-	vhk, err := b.State.UnitVirtualHostKey(unitID)
+// JumpServerVirtualAuthorizedKey returns the authorized_key for the jump server.
+func (b *backend) JumpServerVirtualAuthorizedKey() (string, error) {
+	privKey, err := b.controllerState.SSHServerHostKey()
+	if err != nil {
+		return "", errors.Trace(err)
+	}
+
+	key, err := getAuthorizedKey([]byte(privKey))
 	if err != nil {
 		return "", errors.Trace(err)
 	}
 
-	return string(vhk.HostKey()), nil
+	return key, nil
 }
 
-// MachineVirtualPublicHostKey calls the underlying MachineVirtualHostKey state method
-// and encodes the result into a PEM string.
-func (b *backend) MachineVirtualPublicHostKeyPEM(machineID string) (string, error) {
-	// The keys are persisted PEM encoded.
-	vhk, err := b.State.MachineVirtualHostKey(machineID)
+// UnitVirtualAuthorizedKey returns the authorized_key for the unit.
+func (b *backend) UnitVirtualAuthorizedKey(unitID string) (string, error) {
+	vhk, err := b.controllerState.UnitVirtualHostKey(unitID)
 	if err != nil {
 		return "", errors.Trace(err)
 	}
 
-	return string(vhk.HostKey()), nil
+	key, err := getAuthorizedKey(vhk.HostKey())
+	if err != nil {
+		return "", errors.Trace(err)
+	}
+
+	return key, nil
+}
+
+// MachineVirtualAuthorizedKey returns the authorized_key for the machine.
+func (b *backend) MachineVirtualAuthorizedKey(machineID string) (string, error) {
+	vhk, err := b.controllerState.MachineVirtualHostKey(machineID)
+	if err != nil {
+		return "", errors.Trace(err)
+	}
+
+	key, err := getAuthorizedKey(vhk.HostKey())
+	if err != nil {
+		return "", errors.Trace(err)
+	}
+
+	return key, nil
+}
+
+func getAuthorizedKey(hostkey []byte) (string, error) {
+	signer, err := ssh.ParsePrivateKey(hostkey)
+	if err != nil {
+		return "", errors.Trace(err)
+	}
+
+	pubKey := signer.PublicKey()
+
+	authorizedKey := ssh.MarshalAuthorizedKey(pubKey)
```

Comment:
What about doing `pubKey.Marshal()` method then on the client you can use `ssh.ParsePublicKey` method over the `ssh.ParseAuthorizedKey` function.

Reply:
Marshal returns SSH wire format which is designed to be kept binary. It isn't suitable to be character encoded and must be parsed using "ParsePublicKey". 

Authorized key format is the most suitable of all as it is more general, can be parsed and used in a number of ways.

Reply:
Then what about just base64 encoding it after a call to `Marshal`?

The docstring for MarshalAuthorizedKey() reads,
>// MarshalAuthorizedKey serializes key for inclusion in an OpenSSH
// authorized_keys file. The return value ends with newline.

It's just strange since these are not intended to go into an authorized_keys file so I'm not sure this is the right name or the right function to use for passing the public key around.